packages/react-reconciler/src/ReactFiberRootScheduler.js

```diff
@@ -229,6 +237,12 @@ function processRootScheduleInMicrotask() {
   let root = firstScheduledRoot;
   while (root !== null) {
     const next = root.next;
+    if (includesTransitionLane(root.pendingLanes) && shouldAttemptEagerTransition()) {
```

Comment:
This will force flush transitions even if they were scheduled in a previous event.

We only want to force the transitions if they were scheduled in the current event.

So I think we need to maintain some state like `didEventIncludeTransition` inside this module, and set it to `true` whenever `startTransition` is called. It then gets cleared inside this microtask.

Reply:
We already track something similar in the work loop, called `currentEventTransitionLane`. It currently gets reset at the beginning of `performConcurrentWorkOnRoot` but it would be even better to reset it inside this function.

https://github.com/facebook/react/blob/eeabb7312f509eb2094452a4389646000ea8ea14/packages/react-reconciler/src/ReactFiberWorkLoop.js#L667-L671

So let's move that variable into this module and then we can use it for both purposes.